kernel/xpmem_pfn.c

```diff
@@ -223,7 +227,9 @@ xpmem_pin_page(struct xpmem_thread_group *tg, struct task_struct *src_task,
 	}
 
 	/* get_user_pages() faults and pins the page */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+        ret = get_user_pages(vaddr, 1, FOLL_WRITE|FOLL_FORCE, &page, NULL);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
 	ret = get_user_pages_remote (src_task, src_mm, vaddr, 1, FOLL_WRITE | FOLL_FORCE,
```

Comment:
Did Linux really modify this again? Lots of changes going on in the kernel these days.

Reply:
yes, 4.11 has some substantial changes

Reply:
I guess what is confusing me here is how does it know what task to get the pages from? The new get_user_pages is missing the source task and source mm which is why I moved to using get_user_pages_remote.

Reply:
/* * This is the same as get_user_pages_remote(), just with a *
less-flexible calling convention where we assume that the task * and
mm being operated on are the current task's and don't allow * passing
of a locked parameter.  We also obviously don't pass * FOLL_REMOTE in
here. */